### PR TITLE
[WIP] Re-introduces changes reverted by #453

### PR DESCRIPTION
**Pull request by Xenwar**
Re-introduces changes reverted by #453
Content was reviewed and approved. Therefore, does not need a review again.

This PR by itself will break the ci upgrade tests. Therefore, needs to be merged with other would be created PRs.

### CODE REVIEW

**Comment by maelk** [comment]
@Xenwar So does this need a hold ? I am unclear if it is ready for merge, or if not, what should be waited on.

**Comment by Xenwar** [comment]
> @Xenwar So does this need a hold ? I am unclear if it is ready for merge, or if not, what should be waited on.

Ready for merge, but it needs a hold as well as it will break the ci test runs.
There are other PRs to follow. And, the coming ones and this one need to get merged together.

File: scripts/feature_tests/upgrade/upgrade_common.sh [hunk 1]
```diff
@@ -364,10 +356,10 @@ function cleanup_clusterctl_configuration() {
     mkdir /tmp/cluster-api-clone
 
     # clean up if previous test has failed.
-    rm -rf /home/"${USER}"/.cluster-api/dev-repository/cluster-api/v0.3.8
-    rm -rf /home/"${USER}"/.cluster-api/dev-repository/bootstrap-kubeadm/v0.3.8
-    rm -rf /home/"${USER}"/.cluster-api/dev-repository/control-plane-kubeadm/v0.3.8
-    rm -rf /home/"${USER}"/.cluster-api/overrides/infrastructure-metal3/v0.3.8
```
Comment (Jaakko-Os) [inline scripts/feature_tests/upgrade/upgrade_common.sh:370]: keep these

File: scripts/feature_tests/upgrade/upgrade_common.sh [hunk 2]
```diff
@@ -376,31 +368,31 @@ function create_clusterctl_configuration() {
 cat <<EOF >/home/"${USER}"/.cluster-api/clusterctl.yaml
 providers:
   - name: cluster-api
-    url: /home/$USER/.cluster-api/dev-repository/cluster-api/v0.3.5/core-components.yaml
```
Comment (Jaakko-Os) [inline scripts/feature_tests/upgrade/upgrade_common.sh:379]: General comment: don't change the clusterctl versions and paths. The master branch content is correct.

File: scripts/feature_tests/upgrade/upgrade_common.sh [hunk 3]
```diff
@@ -144,12 +148,10 @@ EOF
 }
 
 function manage_node_taints() {
-    kubectl get secrets "${CLUSTER_NAME}"-kubeconfig -n "${NAMESPACE}" -o json | \
-      jq -r '.data.value'| base64 -d > /tmp/kubeconfig-"${CLUSTER_NAME}".yaml
```
Comment (Jaakko-Os) [inline scripts/feature_tests/upgrade/upgrade_common.sh:148]: General comment: I would do the kubeconfig handling in upgrade_common.sh where it's needed. Will this work after pivoting? I don't have an answer yet.

Like this:

function call
: 
point_to_management_cluster
get_target_kubeconfig
point_to_target_cluster

 // run commands on target cluster
: 
// before exiting
point_to_management_cluster

Comment (Jaakko-Os) [reply scripts/feature_tests/upgrade/upgrade_common.sh:148]: This way the actual test scripts would not need to worry about kubeconfig.

**Review by Jaakko-Os — CHANGES_REQUESTED** [review]

**Comment by Jaakko-Os** [comment]
verification needs to pass with test-upgrade-features and test-upgrade-features-centos before merge

**Comment by Xenwar** [comment]
> verification needs to pass with **test-upgrade-features** and **test-upgrade-features-centos** before merge
**test-upgrade-features** will not pass. 
given the comments, I think we can close this PR and take yours https://github.com/metal3-io/metal3-dev-env/pull/504

File: scripts/feature_tests/upgrade/upgrade_common.sh (hunk 3 above)
Comment (Xenwar) [reply scripts/feature_tests/upgrade/upgrade_common.sh:148]: Sorry, I missed this comment. 
If we take those rows to the common, then how does the test run ?
i.e The logic is in the non-common functions and order varies and cannot be pre-determined in the common functions file

**Pull request closed without merging**